### DISCUSSION TITLE
Will more open-source licenses be accepted other than GNU in the future?

### DISCUSSION THREAD

**Discussion by minsoe** [skiptools/skip#217] [Ideas]
Will other types of open-source licenses be accepted other than GNU in the future?

Hello, I am asking this question because there are a lot of open-source libraries in Swift that can be easily converted to skippable libraries. But, they might not be in GNU licenses, so it can be a blocker to expend the community. Will you remove the limitations of to be GNU license in the future? 

Thanks

**Comment by marcprux** [comment]
The LGPL fits well with the project, and it harmonizes with many of the other projects we have integrations with: [JNA](https://github.com/skiptools/skip-ffi), [WebKit](https://github.com/skiptools/skip-web), [JavaScriptCore](https://github.com/skiptools/skip-script), and others. Note that when a Skip integration framework is created, such as [SkipFirebase](https://github.com/skiptools/skip-web), the integration framework itself can be LGPL, even if the backing libraries on both the iOS and Android sides are a different license (Apache-2.0, in the case of Firebase). They are generally considered compatible.

If there are other projects that can, themselves, be converted *into* Skip libraries, then we may handle those on a case-by-case basis. Note that any license can be used with Skip, but in order to support them in Skip's *free* mode, they need to be a compatible OSS license.

**Reply by marcprux** [reply]
Quick update to this: we've change the license used by Skip's frameworks from the LGPL to the MPL. You can read the announcement at https://github.com/orgs/skiptools/discussions/647
